src/middlewared/middlewared/plugins/chart_releases_linux/validation.py

```diff
@@ -75,20 +79,31 @@ async def validate_question(self, verrors, value, question, var_attr, schema_nam
             dict_attrs = {v['variable']: v for v in schema['attrs']}
             for k in filter(lambda k: k in dict_attrs, value):
                 await self.validate_question(
-                    verrors, value[k], dict_attrs[k], var_attr.attrs[k], f'{schema_name}.{k}',
+                    verrors, value[k], dict_attrs[k], var_attr.attrs[k], f'{schema_name}.{k}', release_data,
                 )
 
         elif schema['type'] == 'list' and value:
             for index, item in enumerate(value):
                 item_index, attr = get_list_item_from_value(item, var_attr)
                 if attr:
                     await self.validate_question(
-                        verrors, item, schema['items'][item_index], attr, f'{schema_name}.{index}'
+                        verrors, item, schema['items'][item_index], attr, f'{schema_name}.{index}', release_data,
                     )
 
         for validator_def in filter(lambda k: k in validation_mapping, schema.get('$ref', [])):
             await self.middleware.call(
                 f'chart.release.validate_{validation_mapping[validator_def]}', verrors, value, question, schema_name,
+                release_data,
             )
 
         return verrors
+
+    @private
+    async def validate_port_available_on_node(self, verrors, value, question, schema_name, release_data):
+        if release_data and value in [p['port'] for p in release_data['used_ports']]:
+            # TODO: This still leaves a case where user has multiple ports in a single app and mixes
+            #  them to the same value however in this case we will still get an error raised by k8s.
+            return
+
+        if value in await self.middleware.call('chart.release.used_ports'):
```

Comment:
What about 80, 135-139, 443, 2049 and all other ports that can be busy?

Reply:
The minimum limit right now is 9000 for Apps. However now that check requires a different approach and we can certainly do that. Please advise if you think we should look into that ?

Reply:
( to clarify where we are looking at ports in use other then Apps )

Reply:
Forgot about that, it's ok then